### ISSUE TITLE
Failed to open file for writing: Input/output error: "/home/root/000"

### ISSUE THREAD

**Issue by nobody93** [bmegli/wifi-scan#28]
I am running Linux, but the file system is read only where the root directory is RO, it seems the program using root directory to write some cache but it is RO. 

The way it writes cache to root forcing user to run the program in root, if I can change / configure writing cache not to root directory, it will not just solve my problem of RO file system, but it will permit to run the program in none-root user which is much desirable, is it possible? Appreciate your advice.

Thank you Bartosz Meglicki, great work.

**Comment by bmegli** [comment]
Hi @nobody93,

Is this related to this repository in some way?

Does it happen when calling one of examples?

Please, provide more info.

**Comment by nobody93** [comment]
> Hi @nobody93,
> 
> Is this related to this repository in some way?
Yes, the source code was from this repository. 

> Does it happen when calling one of examples?
> Please, provide more info.

Here is what I am running running examples/wifi_scan_all.c on my Ubuntu laptop, it cannot perform ./wifi-scan-all wlp2s0 in user mode:

"Operation not permitted". The simplest way is to use sudo.
Unable to get scan data: Operation not permitted

I added exception error handler, when the same program to run my RO file system device, the error was:

Exception: Failed to open file for writing: Input/output error: "/home/root/000"

I am not clear if it tried to write something to root as I saw in my device thus requires root access and permission. Will you provide details why it needs root to run wifi scan, how can it change it cache write location from root directory to other RW directories?

Thank you.

Kind regards
